<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Comparing Two Groups with prolfqua • prolfqua</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Comparing Two Groups with prolfqua">
<meta property="og:description" content="prolfqua">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">prolfqua</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.2.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Comparing2Groups.html">Comparing Two Groups with prolfqua</a>
    </li>
    <li>
      <a href="../articles/CreatingConfigurations.html">Creating prolfqua Configurations</a>
    </li>
    <li>
      <a href="../articles/Modelling2Factors.html">Modelling dataset with two Factors</a>
    </li>
    <li>
      <a href="../articles/QCandSampleSize.html">Quality Control &amp; Sample Size Estimation</a>
    </li>
    <li>
      <a href="../articles/QualityControlAndSampleSizeEstimation.html">QC and Sample Size Estimation</a>
    </li>
    <li>
      <a href="../articles/SimulateData.html">Simulate Peptide level Data</a>
    </li>
    <li>
      <a href="../articles/TestingMissingInference.html">Impute</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/wolski/prolfqua/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><link href="Comparing2Groups_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="Comparing2Groups_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><script src="Comparing2Groups_files/plotly-binding-4.10.4/plotly.js"></script><script src="Comparing2Groups_files/typedarray-0.1/typedarray.min.js"></script><link href="Comparing2Groups_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="Comparing2Groups_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><link href="Comparing2Groups_files/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="Comparing2Groups_files/plotly-main-2.11.1/plotly-latest.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Comparing Two Groups with prolfqua</h1>
                        <h4 data-toc-skip class="author">Witold E.
Wolski</h4>
            
            <h4 data-toc-skip class="date">2024-05-22</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wolski/prolfqua/blob/HEAD/vignettes/Comparing2Groups.Rmd" class="external-link"><code>vignettes/Comparing2Groups.Rmd</code></a></small>
      <div class="hidden name"><code>Comparing2Groups.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="purpose">Purpose<a class="anchor" aria-label="anchor" href="#purpose"></a>
</h2>
<p>This vignette demonstrates how two conditions, e.g., treatment
vs. control, can be compared and the differences statistically tested.
We will again use the Ionstar dataset as an example of an LFQ
experiment. This dataset was preprocessed with the MaxQuant software.
After first examining the data using QC plots and then normalizing the
data, we compare groups of replicates with the different dilutions. The
output of the comparison is the difference in the mean intensities for
quantified proteins (log2 fold-change) in each group, along with
statistical parameters such as degrees of freedom, standard errors,
p-value and the FDR.</p>
</div>
<div class="section level2">
<h2 id="loading-protein-abundances-from-maxquant-proteingroups-txt">Loading protein abundances from MaxQuant proteinGroups.txt<a class="anchor" aria-label="anchor" href="#loading-protein-abundances-from-maxquant-proteingroups-txt"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/wolski/prolfqua" class="external-link">prolfqua</a></span><span class="op">)</span></span></code></pre></div>
<p>Specify the path to the MaxQuant <code>proteinGroups.txt</code> file.
The function <code>tidyMQ_ProteinGroups</code> will read the
<code>proteinGroups.txt</code> file and convert it into a tidy table</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu">prolfqua</span><span class="fu">::</span><span class="fu"><a href="../reference/sim_lfq_data_protein_config.html">sim_lfq_data_protein_config</a></span><span class="op">(</span>Nprot <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">xx</span></span></code></pre></div>
<pre><code><span><span class="co">## $data</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 1,200 × 8</span></span></span>
<span><span class="co">##    sample sampleName group_ isotopeLabel protein_Id abundance qValue nr_peptides</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> A_V1   A_V1       A      light        0EfVhX~39…      20.4      0           2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> A_V1   A_V1       A      light        0m5WN4~60…      16.4      0           7</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> A_V1   A_V1       A      light        0YSKpy~28…      18.2      0           2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> A_V1   A_V1       A      light        3QLHfm~89…      23.0      0           1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> A_V1   A_V1       A      light        3QYop0~75…      23.1      0           7</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> A_V1   A_V1       A      light        76k03k~70…      25.5      0           4</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> A_V1   A_V1       A      light        7cbcrd~73…      27.1      0           2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> A_V1   A_V1       A      light        7QuTub~18…      18.4      0           4</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> A_V1   A_V1       A      light        7soopj~53…      19.5      0           2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> A_V1   A_V1       A      light        7zeekV~71…      20.6      0           8</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1,190 more rows</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $config</span></span>
<span><span class="co">## &lt;AnalysisConfiguration&gt;</span></span>
<span><span class="co">##   Public:</span></span>
<span><span class="co">##     clone: function (deep = FALSE) </span></span>
<span><span class="co">##     initialize: function (analysisTableAnnotation, analysisParameter = AnalysisParameters$new()) </span></span>
<span><span class="co">##     parameter: AnalysisParameters, R6</span></span>
<span><span class="co">##     sep: ~</span></span>
<span><span class="co">##     table: AnalysisTableAnnotation, R6</span></span></code></pre>
<p>Create the <code>LFQData</code> class instance and remove zeros from
data (MaxQuant encodes missing values with zero).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lfqdata</span> <span class="op">&lt;-</span> <span class="fu">prolfqua</span><span class="fu">::</span><span class="va"><a href="../reference/LFQData.html">LFQData</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span><span class="va">xx</span><span class="op">$</span><span class="va">data</span>, <span class="va">xx</span><span class="op">$</span><span class="va">config</span><span class="op">)</span></span>
<span><span class="va">lfqdata</span><span class="op">$</span><span class="fu">remove_small_intensities</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">lfqdata</span><span class="op">$</span><span class="fu">factors</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 12 × 3</span></span></span>
<span><span class="co">##    sample  sampleName group_</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> A_V1    A_V1       A     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> A_V2    A_V2       A     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> A_V3    A_V3       A     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> A_V4    A_V4       A     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> B_V1    B_V1       B     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> B_V2    B_V2       B     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> B_V3    B_V3       B     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> B_V4    B_V4       B     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Ctrl_V1 Ctrl_V1    Ctrl  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Ctrl_V2 Ctrl_V2    Ctrl  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> Ctrl_V3 Ctrl_V3    Ctrl  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> Ctrl_V4 Ctrl_V4    Ctrl</span></span></code></pre>
<p>You can always convert the data into wide format.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lfqdata</span><span class="op">$</span><span class="fu">to_wide</span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">data</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 7</span></span></span>
<span><span class="co">##   protein_Id  isotopeLabel  A_V1  A_V2  A_V3  A_V4  B_V1</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> 0EfVhX~3967 light         20.4  23.8  22.3  21.5  19.1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> 0m5WN4~6025 light         16.4  <span style="color: #BB0000;">NA</span>    16.1  18.2  24.3</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> 0YSKpy~2865 light         18.2  17.7  16.7  17.7  18.2</span></span></code></pre>
<div class="section level3">
<h3 id="visualization-of-not-normalized-data">Visualization of not normalized data<a class="anchor" aria-label="anchor" href="#visualization-of-not-normalized-data"></a>
</h3>
<p>After this first setting up of the analysis we show now how to
normalize the proteins and the effect of normalization. Furthermore we
use some functions to visualize the missing values in our data.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lfqplotter</span> <span class="op">&lt;-</span> <span class="va">lfqdata</span><span class="op">$</span><span class="fu">get_Plotter</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">density_nn</span> <span class="op">&lt;-</span> <span class="va">lfqplotter</span><span class="op">$</span><span class="fu">intensity_distribution_density</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="visualization-of-missing-data">Visualization of missing data<a class="anchor" aria-label="anchor" href="#visualization-of-missing-data"></a>
</h4>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lfqplotter</span><span class="op">$</span><span class="fu">NA_heatmap</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/makeMissingHeatmap-1.png" alt="Heatmap where missing proteins (zero in case of MaxQuant reported intensities), black - missing protein intensities, white - present" width="700"><p class="caption">
Heatmap where missing proteins (zero in case of MaxQuant reported
intensities), black - missing protein intensities, white - present
</p>
</div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lfqdata</span><span class="op">$</span><span class="fu">get_Summariser</span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="fu">plot_missingness_per_group</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/missignessPerGroup-1.png" alt="# of proteins with 0,1,...N missing values" width="700"><p class="caption">
# of proteins with 0,1,…N missing values
</p>
</div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lfqplotter</span><span class="op">$</span><span class="fu">missigness_histogram</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/missignessHistogram-1.png" alt="Intensity distribution of proteins depending on # of missing values" width="700"><p class="caption">
Intensity distribution of proteins depending on # of missing values
</p>
</div>
</div>
<div class="section level4">
<h4 id="computing-standard-deviations-mean-and-cv-">Computing standard deviations, mean and CV.<a class="anchor" aria-label="anchor" href="#computing-standard-deviations-mean-and-cv-"></a>
</h4>
<p>Other important statistics such as coefficient of variation, means
and standard deviations can be easily calculated using the
<code>get_Stats</code> function and visualized with a violin plot.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stats</span> <span class="op">&lt;-</span> <span class="va">lfqdata</span><span class="op">$</span><span class="fu">get_Stats</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "group_"</span></span>
<span><span class="co">## NULL</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stats</span><span class="op">$</span><span class="fu">violin</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/PlotCVDistributions-1.png" alt="Violin plots of CVs in the different groups and among all groups" width="700"><p class="caption">
Violin plots of CVs in the different groups and among all groups
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">prolfqua</span><span class="fu">::</span><span class="fu"><a href="../reference/table_facade.html">table_facade</a></span><span class="op">(</span> <span class="va">stats</span><span class="op">$</span><span class="fu">stats_quantiles</span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">wide</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"quantile of "</span>,<span class="va">stats</span><span class="op">$</span><span class="va">stat</span> <span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>quantile of CV</caption>
<thead><tr class="header">
<th align="right">probs</th>
<th align="right">A</th>
<th align="right">All</th>
<th align="right">B</th>
<th align="right">Ctrl</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">0.10</td>
<td align="right">2.014611</td>
<td align="right">3.451035</td>
<td align="right">1.993591</td>
<td align="right">1.762209</td>
</tr>
<tr class="even">
<td align="right">0.25</td>
<td align="right">2.711702</td>
<td align="right">4.370724</td>
<td align="right">2.956419</td>
<td align="right">2.882970</td>
</tr>
<tr class="odd">
<td align="right">0.50</td>
<td align="right">3.655067</td>
<td align="right">5.296948</td>
<td align="right">3.700140</td>
<td align="right">4.261204</td>
</tr>
<tr class="even">
<td align="right">0.75</td>
<td align="right">5.338899</td>
<td align="right">7.411380</td>
<td align="right">5.462486</td>
<td align="right">5.904953</td>
</tr>
<tr class="odd">
<td align="right">0.90</td>
<td align="right">7.541564</td>
<td align="right">9.912513</td>
<td align="right">7.118907</td>
<td align="right">7.471854</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stats</span><span class="op">$</span><span class="fu">density_median</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/plotCVsplitbyMedianIntensity-1.png" alt="Distribution of CV's for top 50% and bottom 50% proteins by intensity." width="700"><p class="caption">
Distribution of CV’s for top 50% and bottom 50% proteins by intensity.
</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="normalize-protein-intensities-and-show-diagnostic-plots">Normalize protein intensities and show diagnostic plots<a class="anchor" aria-label="anchor" href="#normalize-protein-intensities-and-show-diagnostic-plots"></a>
</h3>
<p>We normalize the data by <span class="math inline">\(\log_2\)</span>
transforming and then <span class="math inline">\(z-scaling\)</span>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lt</span> <span class="op">&lt;-</span> <span class="va">lfqdata</span><span class="op">$</span><span class="fu">get_Transformer</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">transformed</span> <span class="op">&lt;-</span> <span class="va">lt</span><span class="op">$</span><span class="fu">log2</span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="fu">robscale</span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">lfq</span></span>
<span><span class="va">transformed</span><span class="op">$</span><span class="va">config</span><span class="op">$</span><span class="va">table</span><span class="op">$</span><span class="va">is_response_transformed</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">transformed</span><span class="op">$</span><span class="fu">get_Plotter</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">density_norm</span> <span class="op">&lt;-</span> <span class="va">pl</span><span class="op">$</span><span class="fu">intensity_distribution_density</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span><span class="va">density_nn</span>, <span class="va">density_norm</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/showIntensityDistributions-1.png" alt="Distribution of intensities before and after normalization." width="700"><p class="caption">
Distribution of intensities before and after normalization.
</p>
</div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pl</span><span class="op">$</span><span class="fu">pairs_smooth</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/plotScatterMatrix-1.png" alt="Scatterplot matrix" width="700"><p class="caption">
Scatterplot matrix
</p>
</div>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">pl</span><span class="op">$</span><span class="fu">heatmap_cor</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: 5">
<img src="Comparing2Groups_files/figure-html/plotHeatmap-1.png" alt="Heatmap, Rows - proteins, Columns - samples" width="700"><p class="caption">
Heatmap, Rows - proteins, Columns - samples
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="fitting-a-linear-model">Fitting a linear model<a class="anchor" aria-label="anchor" href="#fitting-a-linear-model"></a>
</h2>
<p>For fitting linear models to the transformed intensities for all our
proteins we have to first specify the model function and define the
contrasts that we want to calculate.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">transformed</span><span class="op">$</span><span class="fu">factors</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 12 × 3</span></span></span>
<span><span class="co">##    sample  sampleName group_</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> A_V1    A_V1       A     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> A_V2    A_V2       A     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> A_V3    A_V3       A     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> A_V4    A_V4       A     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> B_V1    B_V1       B     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> B_V2    B_V2       B     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> B_V3    B_V3       B     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> B_V4    B_V4       B     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> Ctrl_V1 Ctrl_V1    Ctrl  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Ctrl_V2 Ctrl_V2    Ctrl  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> Ctrl_V3 Ctrl_V3    Ctrl  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> Ctrl_V4 Ctrl_V4    Ctrl</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">formula_Condition</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/strategy.html">strategy_lm</a></span><span class="op">(</span><span class="st">"transformedIntensity ~ group_"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># specify model definition</span></span>
<span><span class="va">modelName</span>  <span class="op">&lt;-</span> <span class="st">"Model"</span></span>
<span><span class="va">Contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AvsC"</span> <span class="op">=</span> <span class="st">"group_A - group_Ctrl"</span>,</span>
<span><span class="st">"BvsC"</span> <span class="op">=</span> <span class="st">"group_B - group_Ctrl"</span><span class="op">)</span></span></code></pre></div>
<p>Here we have to build the model for each protein.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu">prolfqua</span><span class="fu">::</span><span class="fu"><a href="../reference/build_model.html">build_model</a></span><span class="op">(</span></span>
<span>  <span class="va">transformed</span><span class="op">$</span><span class="va">data</span>,</span>
<span>  <span class="va">formula_Condition</span>,</span>
<span>  subject_Id <span class="op">=</span> <span class="va">transformed</span><span class="op">$</span><span class="va">config</span><span class="op">$</span><span class="va">table</span><span class="op">$</span><span class="fu">hierarchy_keys</span><span class="op">(</span><span class="op">)</span> <span class="op">)</span></span></code></pre></div>
<p>In this plot we can see what factors in our model are mostly
responsible for the adjusted p-values calculated from an analysis of
variance.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mod</span><span class="op">$</span><span class="fu">anova_histogram</span><span class="op">(</span><span class="st">"FDR"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $plot</span></span></code></pre>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/showANOVA-1.png" alt="Distribtuion of adjusted p-values (FDR)" width="700"><p class="caption">
Distribtuion of adjusted p-values (FDR)
</p>
</div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## $name</span></span>
<span><span class="co">## [1] "Anova_p.values_Model.pdf"</span></span></code></pre>
<p>One also look what proteins do show different abundances in any of
our five dilutions by looking at the FDR values of the analysis of
variane.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aovtable</span> <span class="op">&lt;-</span> <span class="va">mod</span><span class="op">$</span><span class="fu">get_anova</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">aovtable</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 10</span></span></span>
<span><span class="co">##   protein_Id  isSingular nrcoef factor    Df  Sum.Sq  Mean.Sq F.value  p.value</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> 0EfVhX~3967 FALSE           3 group_     2 0.062<span style="text-decoration: underline;">6</span>  0.031<span style="text-decoration: underline;">3</span>     5.18  0.041<span style="text-decoration: underline;">6</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> 0m5WN4~6025 FALSE           3 group_     2 0.324   0.162     29.4   0.000<span style="text-decoration: underline;">205</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> 0YSKpy~2865 FALSE           3 group_     2 0.007<span style="text-decoration: underline;">53</span> 0.003<span style="text-decoration: underline;">76</span>    0.605 0.569   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> 3QLHfm~8938 FALSE           3 group_     2 0.063<span style="text-decoration: underline;">7</span>  0.031<span style="text-decoration: underline;">9</span>    13.2   0.002<span style="text-decoration: underline;">13</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> 3QYop0~7543 FALSE           3 group_     2 0.001<span style="text-decoration: underline;">21</span> 0.000<span style="text-decoration: underline;">603</span>   0.178 0.840   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> 76k03k~7094 FALSE           3 group_     2 0.026<span style="text-decoration: underline;">4</span>  0.013<span style="text-decoration: underline;">2</span>     3.06  0.096<span style="text-decoration: underline;">8</span>  </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 1 more variable: FDR &lt;dbl&gt;</span></span></span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">aovtable</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 98 10</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">xx</span> <span class="op">&lt;-</span> <span class="va">aovtable</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="va">signif</span> <span class="op">&lt;-</span> <span class="va">transformed</span><span class="op">$</span><span class="fu">get_copy</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">signif</span><span class="op">$</span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">signif</span><span class="op">$</span><span class="va">data</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">protein_Id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">xx</span><span class="op">$</span><span class="va">protein_Id</span><span class="op">)</span></span>
<span><span class="va">hmSig</span> <span class="op">&lt;-</span> <span class="va">signif</span><span class="op">$</span><span class="fu">get_Plotter</span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="fu">heatmap</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hmSig</span></span></code></pre></div>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/showSigHeatmap-1.png" alt="Heatmap for proteins with FDR &lt; 0.2 in the analysis of variance" width="700"><p class="caption">
Heatmap for proteins with FDR &lt; 0.2 in the analysis of variance
</p>
</div>
</div>
<div class="section level2">
<h2 id="compute-contrasts">Compute contrasts<a class="anchor" aria-label="anchor" href="#compute-contrasts"></a>
</h2>
<p>Next we do calculate the statistics for our defined contrasts for all
the proteins. For this we can use the <code>Contrasts</code>
function.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contr</span> <span class="op">&lt;-</span> <span class="fu">prolfqua</span><span class="fu">::</span><span class="va"><a href="../reference/Contrasts.html">Contrasts</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span><span class="va">mod</span>, <span class="va">Contrasts</span><span class="op">)</span></span>
<span><span class="va">v1</span> <span class="op">&lt;-</span> <span class="va">contr</span><span class="op">$</span><span class="fu">get_Plotter</span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="fu">volcano</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, we can moderate the variance and using the
Experimental Bayes method implemented in
<code>ContrastsModerated</code>.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contr</span> <span class="op">&lt;-</span> <span class="fu">prolfqua</span><span class="fu">::</span><span class="va"><a href="../reference/ContrastsModerated.html">ContrastsModerated</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span><span class="va">contr</span><span class="op">)</span></span>
<span><span class="va">contrdf</span> <span class="op">&lt;-</span> <span class="va">contr</span><span class="op">$</span><span class="fu">get_contrasts</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>In the next figure it can be seen WEWinputNEEDED.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plotter</span> <span class="op">&lt;-</span> <span class="va">contr</span><span class="op">$</span><span class="fu">get_Plotter</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">v2</span> <span class="op">&lt;-</span> <span class="va">plotter</span><span class="op">$</span><span class="fu">volcano</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span><span class="va">v1</span><span class="op">$</span><span class="va">FDR</span>,<span class="va">v2</span><span class="op">$</span><span class="va">FDR</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/plotVolcanos-1.png" alt="Volcano plot, Left panel - no moderation, Right panel - with moderation." width="700"><p class="caption">
Volcano plot, Left panel - no moderation, Right panel - with moderation.
</p>
</div>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plotter</span><span class="op">$</span><span class="fu">ma_plotly</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<div class="plotly html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"data":[{"x":[4.4276081959627236,4.2065906658357948,4.1556302496351849,4.5621289380546077,4.5144353596969262,4.6591450416653117,4.7007339356411899,4.426859784668042,4.3286354579159143,4.3906991163554387,4.3578186030747279,4.1522447737738348,4.3503375519646088,4.6771934319746986,4.5374833537419068,4.0034199083816242,4.2867225057183909,4.8082305646565962,4.3607098434502323,3.9369688473431284,4.0318836195610679,4.2241672503271266,4.3101830942863852,4.3586780577509012,3.8606069128816962,4.2328538203389803,4.1023221415144073,4.3197790043916067,3.9834244304440087,4.1083882193218555,4.3342849826634193,4.5268522350239762,4.1960329374931105,4.9590496129506452,4.026785801371001,4.6262097889405469,4.3747773962973717,4.2651429465075648,4.3468214961409704,4.5413243882109819,4.2139661428116266,4.3834451704396153,4.6316435187189953,4.2220549511844263,4.6725334259795979,4.3130079143532489,4.7328572751676647,4.0581282770946823,4.3220780525005669,4.5224347094136803,4.2447675109782139,3.9705979313466599,4.549311727093623,4.429508177200999,4.4600217272051017,4.175365586795083,4.3470694965001462,4.410438228763951,4.4786083246534236,4.2896843159694074,4.1819150358958002,4.7751501402416654,4.3395432734225192,4.3440488234185404,4.3142912090458942,4.2387972242123402,4.0949151019713463,5.0101385937769241,4.033155931723611,4.3067055721567353,4.5700694980992269,4.7332062813526328,4.0314769385854365,4.9864238776394796,3.9238144390470167,4.5409488658179544,4.5394150429979252,4.4803725844643232,4.5152474706780579,4.2082151410039534,4.3005124452213579,4.2799427522691387,4.4749376281104842,4.3256651561221924,4.2057059999301094,4.0998126308800691,4.0487569206772651,4.2995764711298179,4.0678222183838884,4.3130739563016389,4.418836358951717,4.504070318097531,4.0293620427840544,4.2838955722363714,4.2634455166613447,3.9130146742960101,4.2319374932524525,3.9378897116172289,4.0170575168885838,3.6724499063436924],"y":[0.075784616359704043,-0.26030299517962807,-0.017599105667432079,-0.037308375040460091,-0.021503547362678579,0.022730405270089446,0.095965015724968239,-0.1397438147266031,-0.0015451628674545785,-0.034755794564570001,0.092975891240325403,-0.08451784638037553,-0.032581560750847519,-0.0045163473927437052,0.10688747419997496,-0.014668410227143559,0.050362379423621977,-0.048217560686933239,0.08352210931513479,0.11049877339842447,0.061888732520892081,-0.10654089501919027,0.032238850548893472,-0.040954495333469808,-0.26421333378262685,0.021268229186465287,0.20711565471733195,0.053762196253915366,-0.18555647626346206,0.023744210210288798,-0.067765034312563047,0.17668984873828186,-0.0032993730548023493,-0.019772378815086446,0.028750013693435609,-0.02542985942298787,-0.04237978709039332,0.13107094813232292,0.029594028717393837,-0.13644520432389137,0.063398465318006816,-0.27861673475245585,0.031259750209749637,-0.052170081911812764,-0.21143589820067493,0.038600774511187964,-0.16572394717665992,0.11606976369860419,-0.023937730155632414,0.061359601275431053,0.075713737545016355,0.023533144807325143,-0.0637127106367045,0.034153757310301035,0.012596495929883718,-0.12204527997927016,-0.0037378185605410393,0.018466443083391085,-0.010386053896980895,-0.067690126936401085,0.08746711206560924,-0.057138122423063764,-0.013032695949732968,-0.0020298912124266373,-0.047861532834972527,0.010475063542872146,-0.018453178264132643,-0.14643718899784697,-0.33338170329156858,-0.028848270378495192,-0.030514324188872829,-0.041417032688334944,0.18725249545438788,-0.077859443388300129,0.16248458797656801,-0.11602452784075516,0.065676692889173313,-0.053570443093221329,0.0043986165686160095,-0.097079921583071119,-0.03930267486864028,0.051306557086206503,-0.015908154898447043,-0.012028591840842285,0.036026288593270464,0.08281261844471044,-0.058508567094583519,0.038498517889008731,0.033187756526542406,-0.063104193850233761,-0.046680838139675267,0.048116539604292609,-0.036405334112460379,-0.042408892860454184,0.056995980605768791,0.036625539005184386,-0.10187414003326944,-0.14576231897010811,0.22302562043106886,0.14361294842075534],"text":["0EfVhX~3967","0m5WN4~6025","0YSKpy~2865","3QLHfm~8938","3QYop0~7543","76k03k~7094","7cbcrd~7351","7QuTub~1867","7soopj~5352","7zeekV~7127","9VUkAq~9664","At886V~1021","aWq1bG~0148","bafnr3~8757","BEJI92~2054","bkh7dJ~7252","bzNYQZ~9486","BzYlug~5495","CGzoYe~1884","cp9zgr~5277","cPmoxD~0749","CtOJ9t~0799","cZ8H0x~7340","D7gBzw~6058","dEFcPn~2476","dhTqQ1~8660","DoWup2~8058","drVyKT~6645","DuwH7n~8665","eEoiyG~5605","ErnbC3~2390","Fl4JiV~1049","Fmzxkg~0972","fqkp7n~0643","fuL6vC~0872","FVUJxd~4838","GH4YG0~0964","gjjeAv~3264","GK8UnK~1670","HC8K98~2674","HvIpHG~0003","I1Jk2Z~6072","jaoskV~8712","JcKVfU~5501","JfvT8X~9207","jsxCTN~8371","JV3Z7t~9666","kmZBl5~6461","kuzwUH~2256","lhl9bl~8042","LlQY7I~8419","LMQBev~8107","LpCrO9~2048","lZxnWa~8423","Mfdhc7~5486","MlNn1V~2620","MvYo22~7156","Mwva6m~0521","N1vPxG~6232","nApyU2~5594","NbxDkQ~6565","nKkahQ~6960","OWABXO~0178","OyVSi8~2120","QD5dQ4~0683","QeIC06~3178","QFDgyT~7129","R2i6w7~2550","r2J0Eh~3356","RE5qEz~4581","rPVm8P~0545","RvGeNv~3001","SGIVBl~0253","sUIOl6~2511","TLb4x1~5613","uJeKb5~5619","v2Ck8h~4921","VUnZ6o~6572","WjJPCz~5510","WYeN84~8562","ydeWJl~0020","ZDkJUz~5950","zwajUl~8881","zYsdc2~8076","b5eKpF~2415","FbZ3Nt~6818","hoePxF~6252","LyXRwS~4992","oXGLMQ~9289","PP9CBP~5509","psyuFD~5447","qdJkdz~4494","umjHSH~2090","wcxs7a~4833","WyMJ2f~3635","ZI2MGy~7309","ZX1MgP~7534","cnCfON~9769","McjZyZ~3638","ZHBkZv~0394"],"key":["0EfVhX~3967","0m5WN4~6025","0YSKpy~2865","3QLHfm~8938","3QYop0~7543","76k03k~7094","7cbcrd~7351","7QuTub~1867","7soopj~5352","7zeekV~7127","9VUkAq~9664","At886V~1021","aWq1bG~0148","bafnr3~8757","BEJI92~2054","bkh7dJ~7252","bzNYQZ~9486","BzYlug~5495","CGzoYe~1884","cp9zgr~5277","cPmoxD~0749","CtOJ9t~0799","cZ8H0x~7340","D7gBzw~6058","dEFcPn~2476","dhTqQ1~8660","DoWup2~8058","drVyKT~6645","DuwH7n~8665","eEoiyG~5605","ErnbC3~2390","Fl4JiV~1049","Fmzxkg~0972","fqkp7n~0643","fuL6vC~0872","FVUJxd~4838","GH4YG0~0964","gjjeAv~3264","GK8UnK~1670","HC8K98~2674","HvIpHG~0003","I1Jk2Z~6072","jaoskV~8712","JcKVfU~5501","JfvT8X~9207","jsxCTN~8371","JV3Z7t~9666","kmZBl5~6461","kuzwUH~2256","lhl9bl~8042","LlQY7I~8419","LMQBev~8107","LpCrO9~2048","lZxnWa~8423","Mfdhc7~5486","MlNn1V~2620","MvYo22~7156","Mwva6m~0521","N1vPxG~6232","nApyU2~5594","NbxDkQ~6565","nKkahQ~6960","OWABXO~0178","OyVSi8~2120","QD5dQ4~0683","QeIC06~3178","QFDgyT~7129","R2i6w7~2550","r2J0Eh~3356","RE5qEz~4581","rPVm8P~0545","RvGeNv~3001","SGIVBl~0253","sUIOl6~2511","TLb4x1~5613","uJeKb5~5619","v2Ck8h~4921","VUnZ6o~6572","WjJPCz~5510","WYeN84~8562","ydeWJl~0020","ZDkJUz~5950","zwajUl~8881","zYsdc2~8076","b5eKpF~2415","FbZ3Nt~6818","hoePxF~6252","LyXRwS~4992","oXGLMQ~9289","PP9CBP~5509","psyuFD~5447","qdJkdz~4494","umjHSH~2090","wcxs7a~4833","WyMJ2f~3635","ZI2MGy~7309","ZX1MgP~7534","cnCfON~9769","McjZyZ~3638","ZHBkZv~0394"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,0,0,1)","opacity":0.5,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,0,0,1)"}},"hoveron":"points","set":"SharedData892c07d8","name":"WaldTest_moderated","legendgroup":"WaldTest_moderated","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[4.3321475392198838,4.4237370299600984,4.1331836126694528,4.4958644732325004,4.5249418983511349,4.6047164663868241,4.5858544637679364,4.5366195444365749,4.2905374400164655,4.3851289343160982,4.3983125355724253,4.3259637129284867,4.3557770441542045,4.6191265579097411,4.4243912438514332,3.9831562109823202,4.2962409313429397,4.7876326482863316,4.4154232939244578,3.8984918118123861,4.0183471371314887,4.3849169869520574,4.2635126567904988,4.3136687979794486,3.9140296722128771,4.1739484928894619,3.9531632408817186,4.3926140731869188,4.2080777168666375,4.1145915253035801,4.3575698953675648,4.5454841416452778,4.2093636487432562,4.9067636602768445,4.0171704217843205,4.7064803242164075,4.3480320043971936,4.2356075865301026,4.274827130564546,4.6807239897203061,4.1111307946566047,4.5874147927841022,4.5975249847145658,4.2710885768721552,4.785852203375609,4.280298266983567,4.8561410530942162,4.0273874578337043,4.4121696086759137,4.4799380690325776,4.1847853453918438,3.975951881867398,4.5699905906902307,4.4293816087318252,4.439384163780673,4.1787985730520187,4.3065577474314267,4.4058684133237982,4.4828833098678711,4.2832990232674124,4.1611210330294801,4.7236933621300254,4.3200442181858447,4.3217462417737593,4.2827965405993789,4.2722936669785065,4.0692178796756657,5.0925875195634882,4.3220805294021467,4.273226065524395,4.5399471846388542,4.6734433624204472,4.0880577131364539,4.9469163283082569,3.8906329812272888,4.5327033629971902,4.4976223427581443,4.4504039732454768,4.4620304502669983,4.1688593949610615,4.2978703717680942,4.2524794079354358,4.4900169204591247,4.3295809046104141,4.1370864703620267,4.0787260735370108,4.0550031530286699,4.3014983400280427,4.0232899866501075,4.2911290838555987,4.4221997741297008,4.4221617466416463,4.0476602632677379,4.3207467626145579,4.1995786288324304,3.8312914036565884,4.1900155919583355,3.9641273476980583,4.0857504292603526,3.6822860801509969],"y":[-0.1151366971259761,0.17398973306897941,-0.062492379598896954,-0.16983730468467509,-0.00049047005426073459,-0.086126745286886616,-0.13379392802153903,0.079775704810462728,-0.077741198666351471,-0.04589615864325107,0.17396375623572036,0.26292003192892754,-0.021702576371655392,-0.1206500955226593,-0.11929674558097217,-0.055195805025750951,0.069399230672719581,-0.089413393427462781,0.19294901026358618,0.033544702336940055,0.034815767661734394,0.21495857823067088,-0.061102024442879738,-0.13097301487637525,-0.15736781512026532,-0.096542425712571323,-0.091202146548047147,0.19943233384453971,0.26375009658179527,0.03615082217373744,-0.021195208904272393,0.21395366198088572,0.023362049445489797,-0.12434428416268718,0.0095192545200748387,0.13511121112873295,-0.095870570890749032,0.072000228177398831,-0.11439470243545574,0.14235399869475654,-0.14227223099203734,0.12932250993651739,-0.036977317799109677,0.045897169463645159,0.015201656591348034,-0.026818520228176539,0.080843608676443024,0.054588125176647753,0.15624538219506129,-0.023633679486773446,-0.044250593627724169,0.034241045848800944,-0.022354983443488394,0.033900620371953405,-0.028678630918973706,-0.11517930746539948,-0.084761316697980607,0.0093268122030851826,-0.0018360834680852187,-0.080460712340390414,0.045879106332969634,-0.1600516786463444,-0.052030806423082325,-0.046635054501989098,-0.11085086972800268,0.077467949075205156,-0.069847622855494507,0.018460662575282172,0.244467492065504,-0.095807283643176699,-0.090758951109618707,-0.16094287055270565,0.30041404455642245,-0.15687454205074558,0.096121672337112052,-0.13251553348228273,-0.017908707590387651,-0.11350766553091352,-0.10203542425350326,-0.17579141366885448,-0.044586821775167415,-0.0036201315811993581,0.01425042979883372,-0.0041970948643996805,-0.10121277054289536,0.040639503758594646,-0.046016102391775657,0.042342255685457561,-0.055876706941018789,-0.10699393874231466,-0.039954007783707959,-0.11570060330747692,0.00019110685490652396,0.031293487895919542,-0.070737795052060626,-0.12682100227365878,-0.18571794262150415,-0.0932870468084495,0.36041144517460666,0.16328529603536474],"text":["0EfVhX~3967","0m5WN4~6025","0YSKpy~2865","3QLHfm~8938","3QYop0~7543","76k03k~7094","7cbcrd~7351","7QuTub~1867","7soopj~5352","7zeekV~7127","9VUkAq~9664","At886V~1021","aWq1bG~0148","bafnr3~8757","BEJI92~2054","bkh7dJ~7252","bzNYQZ~9486","BzYlug~5495","CGzoYe~1884","cp9zgr~5277","cPmoxD~0749","CtOJ9t~0799","cZ8H0x~7340","D7gBzw~6058","dEFcPn~2476","dhTqQ1~8660","DoWup2~8058","drVyKT~6645","DuwH7n~8665","eEoiyG~5605","ErnbC3~2390","Fl4JiV~1049","Fmzxkg~0972","fqkp7n~0643","fuL6vC~0872","FVUJxd~4838","GH4YG0~0964","gjjeAv~3264","GK8UnK~1670","HC8K98~2674","HvIpHG~0003","I1Jk2Z~6072","jaoskV~8712","JcKVfU~5501","JfvT8X~9207","jsxCTN~8371","JV3Z7t~9666","kmZBl5~6461","kuzwUH~2256","lhl9bl~8042","LlQY7I~8419","LMQBev~8107","LpCrO9~2048","lZxnWa~8423","Mfdhc7~5486","MlNn1V~2620","MvYo22~7156","Mwva6m~0521","N1vPxG~6232","nApyU2~5594","NbxDkQ~6565","nKkahQ~6960","OWABXO~0178","OyVSi8~2120","QD5dQ4~0683","QeIC06~3178","QFDgyT~7129","R2i6w7~2550","r2J0Eh~3356","RE5qEz~4581","rPVm8P~0545","RvGeNv~3001","SGIVBl~0253","sUIOl6~2511","TLb4x1~5613","uJeKb5~5619","v2Ck8h~4921","VUnZ6o~6572","WjJPCz~5510","WYeN84~8562","ydeWJl~0020","ZDkJUz~5950","zwajUl~8881","zYsdc2~8076","b5eKpF~2415","FbZ3Nt~6818","hoePxF~6252","LyXRwS~4992","oXGLMQ~9289","PP9CBP~5509","psyuFD~5447","qdJkdz~4494","umjHSH~2090","wcxs7a~4833","WyMJ2f~3635","ZI2MGy~7309","ZX1MgP~7534","cnCfON~9769","McjZyZ~3638","ZHBkZv~0394"],"key":["0EfVhX~3967","0m5WN4~6025","0YSKpy~2865","3QLHfm~8938","3QYop0~7543","76k03k~7094","7cbcrd~7351","7QuTub~1867","7soopj~5352","7zeekV~7127","9VUkAq~9664","At886V~1021","aWq1bG~0148","bafnr3~8757","BEJI92~2054","bkh7dJ~7252","bzNYQZ~9486","BzYlug~5495","CGzoYe~1884","cp9zgr~5277","cPmoxD~0749","CtOJ9t~0799","cZ8H0x~7340","D7gBzw~6058","dEFcPn~2476","dhTqQ1~8660","DoWup2~8058","drVyKT~6645","DuwH7n~8665","eEoiyG~5605","ErnbC3~2390","Fl4JiV~1049","Fmzxkg~0972","fqkp7n~0643","fuL6vC~0872","FVUJxd~4838","GH4YG0~0964","gjjeAv~3264","GK8UnK~1670","HC8K98~2674","HvIpHG~0003","I1Jk2Z~6072","jaoskV~8712","JcKVfU~5501","JfvT8X~9207","jsxCTN~8371","JV3Z7t~9666","kmZBl5~6461","kuzwUH~2256","lhl9bl~8042","LlQY7I~8419","LMQBev~8107","LpCrO9~2048","lZxnWa~8423","Mfdhc7~5486","MlNn1V~2620","MvYo22~7156","Mwva6m~0521","N1vPxG~6232","nApyU2~5594","NbxDkQ~6565","nKkahQ~6960","OWABXO~0178","OyVSi8~2120","QD5dQ4~0683","QeIC06~3178","QFDgyT~7129","R2i6w7~2550","r2J0Eh~3356","RE5qEz~4581","rPVm8P~0545","RvGeNv~3001","SGIVBl~0253","sUIOl6~2511","TLb4x1~5613","uJeKb5~5619","v2Ck8h~4921","VUnZ6o~6572","WjJPCz~5510","WYeN84~8562","ydeWJl~0020","ZDkJUz~5950","zwajUl~8881","zYsdc2~8076","b5eKpF~2415","FbZ3Nt~6818","hoePxF~6252","LyXRwS~4992","oXGLMQ~9289","PP9CBP~5509","psyuFD~5447","qdJkdz~4494","umjHSH~2090","wcxs7a~4833","WyMJ2f~3635","ZI2MGy~7309","ZX1MgP~7534","cnCfON~9769","McjZyZ~3638","ZHBkZv~0394"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,0,0,1)","opacity":0.5,"size":5.6692913385826778,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,0,0,1)"}},"hoveron":"points","set":"SharedData892c07d8","name":"WaldTest_moderated","legendgroup":"WaldTest_moderated","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[3.6014430256827028,5.1635944002244782,null,3.6014430256827028,5.1635944002244782],"y":[-1,-1,null,1,1],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(255,0,0,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[3.6014430256827028,5.1635944002244782,null,3.6014430256827028,5.1635944002244782],"y":[-1,-1,null,1,1],"text":"","type":"scatter","mode":"lines","line":{"width":1.8897637795275593,"color":"rgba(255,0,0,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":39.514134135719381,"r":7.3059360730593621,"b":41.778974318367787,"l":48.949771689497723},"plot_bgcolor":"rgba(235,235,235,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,0.49608610567514677],"automargin":true,"type":"linear","autorange":false,"range":[3.6014430256827028,5.1635944002244782],"tickmode":"array","ticktext":["4.0","4.4","4.8"],"tickvals":[4,4.4000000000000004,4.8000000000000007],"categoryorder":"array","categoryarray":["4.0","4.4","4.8"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"annotations":[{"text":"avgAbd","x":0.5,"y":0,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"top","annotationType":"axis","yshift":-21.917808219178081},{"text":"diff","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-33.607305936073068},{"text":"AvsC","x":0.24804305283757339,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.68949771689498},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"BvsC","x":0.75195694716242656,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(26,26,26,1)","family":"","size":11.68949771689498},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1.1000000000000001,1.1000000000000001],"tickmode":"array","ticktext":["-1.0","-0.5","0.0","0.5","1.0"],"tickvals":[-1,-0.5,0,0.5,1],"categoryorder":"array","categoryarray":["-1.0","-0.5","0.0","0.5","1.0"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":0.49608610567514677,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(217,217,217,1)","line":{"color":"transparent","width":0.66417600664176002,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":0.49608610567514677,"y0":0,"y1":23.37899543378996,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0.50391389432485323,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":"rgba(217,217,217,1)","line":{"color":"transparent","width":0.66417600664176002,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0.50391389432485323,"x1":1,"y0":0,"y1":23.37899543378996,"yanchor":1,"ysizemode":"pixel"}],"xaxis2":{"type":"linear","autorange":false,"range":[3.6014430256827028,5.1635944002244782],"tickmode":"array","ticktext":["4.0","4.4","4.8"],"tickvals":[4,4.4000000000000004,4.8000000000000007],"categoryorder":"array","categoryarray":["4.0","4.4","4.8"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"domain":[0.50391389432485323,1],"gridcolor":"rgba(255,255,255,1)","gridwidth":0.66417600664176002,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498},"title":{"text":"modelName","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724}}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"129755e8e97a4":{"x":{},"y":{},"text":{},"colour":{},"type":"scatter"},"129756385c07b":{"yintercept":{}}},"cur_data":"129755e8e97a4","visdat":{"129755e8e97a4":["function (y) ","x"],"129756385c07b":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData892c07d8"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script><p class="caption">
MA plot showing the dependency of mean abuncance with respect to the
difference
</p>
</div>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#myProteinIDS &lt;- c("sp|Q12246|LCB4_YEAST",  "sp|P38929|ATC2_YEAST",  "sp|Q99207|NOP14_YEAST")</span></span>
<span><span class="va">myProteinIDS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sp|P0AC33|FUMA_ECOLI"</span>,  <span class="st">"sp|P28635|METQ_ECOLI"</span>,  <span class="st">"sp|Q14C86|GAPD1_HUMAN"</span><span class="op">)</span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">contrdf</span>, <span class="va">protein_Id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">myProteinIDS</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 0 × 13</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 13 variables: modelName &lt;chr&gt;, protein_Id &lt;chr&gt;, contrast &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   diff &lt;dbl&gt;, std.error &lt;dbl&gt;, avgAbd &lt;dbl&gt;, statistic &lt;dbl&gt;, df &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   p.value &lt;dbl&gt;, conf.low &lt;dbl&gt;, conf.high &lt;dbl&gt;, sigma &lt;dbl&gt;, FDR &lt;dbl&gt;</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="contrasts-with-missing-value-imputation">Contrasts with missing value imputation<a class="anchor" aria-label="anchor" href="#contrasts-with-missing-value-imputation"></a>
</h2>
<p>Use this method if there proteins with no observations in one of the
groups. With the <code>ContrastsMissing</code> function, we can estimate
difference in mean for proteins that are not observed in one group or
condition. For this we are using the average expression at percentile
0.05 of the group where the protein is not quantified.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mC</span> <span class="op">&lt;-</span> <span class="va"><a href="../reference/ContrastsMissing.html">ContrastsMissing</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>lfqdata <span class="op">=</span> <span class="va">transformed</span>, contrasts <span class="op">=</span> <span class="va">Contrasts</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mC</span><span class="op">$</span><span class="fu">get_contrasts</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "group_"</span></span></code></pre>
<pre><code><span><span class="co">##  [1] "modelName"                "protein_Id"              </span></span>
<span><span class="co">##  [3] "meanAbundanceImp_group_1" "meanAbundanceImp_group_2"</span></span>
<span><span class="co">##  [5] "diff"                     "group_1_name"            </span></span>
<span><span class="co">##  [7] "group_2_name"             "contrast"                </span></span>
<span><span class="co">##  [9] "avgAbd"                   "indic"                   </span></span>
<span><span class="co">## [11] "nrMeasured_group_1"       "nrMeasured_group_2"      </span></span>
<span><span class="co">## [13] "df"                       "sigma"                   </span></span>
<span><span class="co">## [15] "std.error"                "statistic"               </span></span>
<span><span class="co">## [17] "p.value"                  "conf.low"                </span></span>
<span><span class="co">## [19] "conf.high"                "FDR"</span></span></code></pre>
<p>Finally we are merging the results and give priority to the results
where we do not have missing values in one group.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merged</span> <span class="op">&lt;-</span> <span class="fu">prolfqua</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_contrasts_results.html">merge_contrasts_results</a></span><span class="op">(</span>prefer <span class="op">=</span> <span class="va">contr</span>,add <span class="op">=</span> <span class="va">mC</span><span class="op">)</span><span class="op">$</span><span class="va">merged</span></span>
<span><span class="va">plotter</span> <span class="op">&lt;-</span> <span class="va">merged</span><span class="op">$</span><span class="fu">get_Plotter</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="va">plotter</span><span class="op">$</span><span class="fu">volcano</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tmp</span><span class="op">$</span><span class="va">FDR</span></span></code></pre></div>
<div class="figure">
<img src="Comparing2Groups_files/figure-html/mergeResults-1.png" alt="Volcano plots for the two contrasts with missing value imputation from the group_average model." width="700"><p class="caption">
Volcano plots for the two contrasts with missing value imputation from
the group_average model.
</p>
</div>
<p>Look at proteins which could not be fitted using the linear model, if
any.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merged</span> <span class="op">&lt;-</span> <span class="fu">prolfqua</span><span class="fu">::</span><span class="fu"><a href="../reference/merge_contrasts_results.html">merge_contrasts_results</a></span><span class="op">(</span>prefer <span class="op">=</span> <span class="va">contr</span>,add <span class="op">=</span> <span class="va">mC</span><span class="op">)</span></span>
<span></span>
<span><span class="va">moreProt</span> <span class="op">&lt;-</span> <span class="va">transformed</span><span class="op">$</span><span class="fu">get_copy</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">moreProt</span><span class="op">$</span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">moreProt</span><span class="op">$</span><span class="va">data</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">protein_Id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">merged</span><span class="op">$</span><span class="va">more</span><span class="op">$</span><span class="va">contrast_result</span><span class="op">$</span><span class="va">protein_Id</span><span class="op">)</span></span>
<span><span class="va">moreProt</span><span class="op">$</span><span class="fu">get_Plotter</span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="fu">raster</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="Comparing2Groups_files/figure-html/mergedMore-1.png" width="700"></p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># here we do not get anything because there is nothing imputed!</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="gsea-analyis">GSEA Analyis<a class="anchor" aria-label="anchor" href="#gsea-analyis"></a>
</h2>
<p>We can rank the proteins based on the log2FC or the t-statistic and
subject them them to gene set enrichment analysis (see GSEA).</p>
<p>This example will run only if the following packages are installed on
you machine:</p>
<ul>
<li>clusterProfiler (Bioconductor)</li>
<li>org.Sc.sgd.db (Bioconductor)</li>
<li>prora (github.com/protviz/prora)</li>
</ul>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#evalAll &lt;- require("clusterProfiler") &amp; require("org.Sc.sgd.db") &amp; require("prora")</span></span>
<span><span class="va">evalAll</span> <span class="op">&lt;-</span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"clusterProfiler"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"org.Sc.sgd.db2"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"prora"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">clusterProfiler</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Sc.sgd.db</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bb</span> <span class="op">&lt;-</span> <span class="fu">prolfqua</span><span class="fu">::</span><span class="fu"><a href="../reference/get_UniprotID_from_fasta_header.html">get_UniprotID_from_fasta_header</a></span><span class="op">(</span><span class="va">merged</span><span class="op">$</span><span class="va">merged</span><span class="op">$</span><span class="fu">get_contrasts</span><span class="op">(</span><span class="op">)</span>,</span>
<span>                                             idcolumn <span class="op">=</span> <span class="st">"protein_Id"</span><span class="op">)</span></span>
<span><span class="va">bb</span> <span class="op">&lt;-</span> <span class="fu">prora</span><span class="fu">::</span><span class="fu">map_ids_uniprot</span><span class="op">(</span><span class="va">bb</span><span class="op">)</span></span>
<span><span class="va">ranklist</span> <span class="op">&lt;-</span> <span class="va">bb</span><span class="op">$</span><span class="va">statistic</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ranklist</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">bb</span><span class="op">$</span><span class="va">P_ENTREZGENEID</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">clusterProfiler</span><span class="fu">::</span><span class="fu">gseGO</span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">ranklist</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>  OrgDb <span class="op">=</span> <span class="va">org.Sc.sgd.db</span>,</span>
<span>  ont <span class="op">=</span> <span class="st">"ALL"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ridgeplot</span><span class="op">(</span> <span class="va">res</span> <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dotplot</span><span class="op">(</span><span class="va">res</span> , showCategory <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">enrichplot</span><span class="fu">::</span><span class="fu">upsetplot</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre></div>
<p>The <code>prolfqua</code> package is described in <span class="citation">(Wolski et al. 2022)</span>.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Sonoma 14.4.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Zurich</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] prolfqua_1.2.5</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] gtable_0.3.5       xfun_0.43          bslib_0.7.0        ggplot2_3.5.1     </span></span>
<span><span class="co">##  [5] htmlwidgets_1.6.4  ggrepel_0.9.5      vctrs_0.6.5        tools_4.3.2       </span></span>
<span><span class="co">##  [9] crosstalk_1.2.1    generics_0.1.3     tibble_3.2.1       fansi_1.0.6       </span></span>
<span><span class="co">## [13] highr_0.10         pkgconfig_2.0.3    pheatmap_1.0.12    KernSmooth_2.23-22</span></span>
<span><span class="co">## [17] data.table_1.15.4  RColorBrewer_1.1-3 desc_1.4.3         lifecycle_1.0.4   </span></span>
<span><span class="co">## [21] compiler_4.3.2     farver_2.1.1       stringr_1.5.1      textshaping_0.3.7 </span></span>
<span><span class="co">## [25] progress_1.2.3     munsell_0.5.1      httpuv_1.6.15      htmltools_0.5.8.1 </span></span>
<span><span class="co">## [29] sass_0.4.9         yaml_2.3.8         lazyeval_0.2.2     plotly_4.10.4     </span></span>
<span><span class="co">## [33] later_1.3.2        pillar_1.9.0       pkgdown_2.0.9      crayon_1.5.2      </span></span>
<span><span class="co">## [37] jquerylib_0.1.4    tidyr_1.3.1        MASS_7.3-60.0.1    cachem_1.0.8      </span></span>
<span><span class="co">## [41] mime_0.12          tidyselect_1.2.1   conflicted_1.2.0   digest_0.6.35     </span></span>
<span><span class="co">## [45] stringi_1.8.3      dplyr_1.1.4        purrr_1.0.2        labeling_0.4.3    </span></span>
<span><span class="co">## [49] forcats_1.0.0      fastmap_1.1.1      grid_4.3.2         colorspace_2.1-0  </span></span>
<span><span class="co">## [53] cli_3.6.2          magrittr_2.0.3     utf8_1.2.4         withr_3.0.0       </span></span>
<span><span class="co">## [57] promises_1.3.0     prettyunits_1.2.0  scales_1.3.0       rmarkdown_2.26    </span></span>
<span><span class="co">## [61] httr_1.4.7         gridExtra_2.3      ragg_1.3.0         hms_1.1.3         </span></span>
<span><span class="co">## [65] shiny_1.8.1.1      memoise_2.0.1      evaluate_0.23      knitr_1.46        </span></span>
<span><span class="co">## [69] viridisLite_0.4.2  rlang_1.1.3        Rcpp_1.0.12        xtable_1.8-4      </span></span>
<span><span class="co">## [73] glue_1.7.0         jsonlite_1.8.8     R6_2.5.1           systemfonts_1.0.6 </span></span>
<span><span class="co">## [77] fs_1.6.4</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Wolski2022.06.07.494524" class="csl-entry">
Wolski, Witold E., Paolo Nanni, Jonas Grossmann, Maria d’Errico, Ralph
Schlapbach, and Christian Panse. 2022. <span>“Prolfqua: A Comprehensive
<span class="nocase">R-package</span> for Proteomics Differential
Expression Analysis.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/2022.06.07.494524" class="external-link">https://doi.org/10.1101/2022.06.07.494524</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Witold Wolski.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
